superset-frontend/src/components/TooltipParagraph/index.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from 'react';
+import { Tooltip } from 'antd';
+import Paragraph, { ParagraphProps } from 'antd/es/typography/Paragraph';
+
+const TooltipParagraph: React.FC<ParagraphProps> = ({
+  children,
+  ellipsis,
+  ...props
+}) => {
+  const [truncated, setTruncated] = useState(false);
+
+  return (
+    <Tooltip title={truncated ? children : undefined}>
+      <Paragraph
+        {...props}
+        ellipsis={{ ...(ellipsis as any), onEllipsis: setTruncated }}
+      >
+        {/* NOTE: Fragment is necessary to avoid showing the title */}
+        <>{children}</>
+      </Paragraph>
+    </Tooltip>
+  );
+};
+
+export default TooltipParagraph;
```

Comment:
If we're going to keep this component, which has value, let's add a storybook entry for it, and at least basic unit/RTL tests.